File: Reverse _linked_list.c
#include <stdio.h>
#include <stdlib.h>

struct Node{
    int data;
    struct Node* next;
};

struct Node* head=NULL;

void insert_end(int val){
    struct Node* newnode=malloc(sizeof(struct Node));
    newnode->data=val; newnode->next=NULL;
    if(!head){head=newnode;return;}
    struct Node* temp=head;
    while(temp->next)temp=temp->next;
    temp->next=newnode;
}

void reverse(){
    struct Node *prev=NULL,*curr=head,*next=NULL;
    while(curr){
        next=curr->next;
        curr->next=prev;
        prev=curr;
        curr=next;
    }
    head=prev;
}

void display(){
    struct Node* temp=head;
    while(temp){printf("%d ",temp->data); temp=temp->next;}
    printf("\n");
}

int main(){
    insert_end(10); insert_end(20); insert_end(30); insert_end(40);
    reverse();
    display();
    return 0;
}

File: Reverse _linked_list.py
class Node:
    def __init__(self,data):
        self.data=data
        self.next=None

class LinkedList:
    def __init__(self):
        self.head=None

    def insert_end(self,val):
        newnode=Node(val)
        if not self.head:self.head=newnode;return
        temp=self.head
        while temp.next: temp=temp.next
        temp.next=newnode

    def reverse(self):
        prev=None
        curr=self.head
        while curr:
            nxt=curr.next
            curr.next=prev
            prev=curr
            curr=nxt
        self.head=prev

    def display(self):
        temp=self.head
        while temp: print(temp.data,end=" "); temp=temp.next
        print()

ll=LinkedList()
ll.insert_end(10); ll.insert_end(20); ll.insert_end(30); ll.insert_end(40)
ll.reverse()
ll.display()
